scripts/check-typescript-compatibility.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { printLog, runMain } = require('./lib/executionUtils')
+const { command } = require('./lib/command')
+
+const TEST_APP_DIR = path.join(__dirname, '..', 'test', 'app')
+
+runMain(() => {
+  printLog('Building project...')
+  command`yarn build`.run()
+
+  printLog('Setting up test environment...')
+  command`yarn`.withCurrentWorkingDirectory(TEST_APP_DIR).run()
+
+  const checks = [
+    {
+      title: 'TypeScript 3.8.2 compatibility',
+      version: '3.8.2',
+    },
+    {
+      title: 'TypeScript isolated modules compatibility',
+      compilerOptions: { isolatedModules: true },
+      version: '3.8.2',
```

Comment:
We used to test `isolatedModules` with 3.8.2, I didn't want to change this without a good reason.